fml/trace_event.cc

```diff
@@ -4,23 +4,36 @@
 
 #include "flutter/fml/trace_event.h"
 
-#include "third_party/dart/runtime/include/dart_tools_api.h"
+#include "flutter/fml/logging.h"
 
 namespace fml {
 namespace tracing {
 
-void TraceCounter(TraceArg category_group, TraceArg name, TraceIDArg count) {
-  auto count_string = std::to_string(count);
-  const char* arg_names[] = {name};
-  const char* arg_values[] = {count_string.c_str()};
-  Dart_TimelineEvent(name,                         // label
-                     Dart_TimelineGetMicros(),     // timestamp0
-                     0,                            // timestamp1_or_async_id
-                     Dart_Timeline_Event_Counter,  // event type
-                     1,                            // argument_count
-                     arg_names,                    // argument_names
-                     arg_values                    // argument_values
+void TraceTimelineEvent(TraceArg category_group,
+                        TraceArg name,
+                        TraceIDArg identifier,
+                        Dart_Timeline_Event_Type type,
+                        const std::vector<const char*>& names,
+                        const std::vector<std::string>& values) {
+  const auto argument_count = std::min(names.size(), values.size());
+
+  auto c_values =
+      static_cast<const char**>(calloc(argument_count, sizeof(const char*)));
```

Comment:
Can you use a std::vector<const char*> here?

Reply:
Yeah.

Reply:
Done